### PR TITLE
Modification of mamltrainer in the branch of `Contest/v1.0`

### PR DESCRIPTION
**Pull request by DavdGao**
@wanghh7 , please refer to the modification in mamltrainer here
PS: we add a library `learn2learn` to support maml in FS, so execute the following cmd to install it
```
pip install learn2learn
```

### CODE REVIEW

File: federatedscope/gfl/trainer/mamltrainer.py [hunk 1]
```diff
@@ -1,48 +1,68 @@
 from federatedscope.register import register_trainer
 from federatedscope.core.trainers.trainer import GeneralTorchTrainer
 
+import learn2learn as l2l
 
 class GraphMAMLTrainer(GeneralTorchTrainer):
+    def _hook_on_fit_start_init(self, ctx):
+        super()._hook_on_fit_start_init(ctx)
+        maml = l2l.algorithms.MAML(self.ctx.model, lr=self.cfg.maml.inner_lr)
+        ctx.clone = maml.clone()
```
Comment (yxdyc) [inline federatedscope/gfl/trainer/mamltrainer.py:10]: Looks a bit strange `ctx.clone`, how about ctx.maml?

File: federatedscope/gfl/trainer/mamltrainer.py [hunk 2]
```diff
@@ -1,48 +1,68 @@
 from federatedscope.register import register_trainer
 from federatedscope.core.trainers.trainer import GeneralTorchTrainer
 
+import learn2learn as l2l
```
Comment (yxdyc) [inline federatedscope/gfl/trainer/mamltrainer.py:4]: Is the new env package in the requirment.txt or the dockerfile?

File: federatedscope/gfl/trainer/mamltrainer.py [hunk 3]
```diff
@@ -1,48 +1,68 @@
...
 class GraphMAMLTrainer(GeneralTorchTrainer):
+    def _hook_on_fit_start_init(self, ctx):
+        super()._hook_on_fit_start_init(ctx)
+        maml = l2l.algorithms.MAML(self.ctx.model, lr=self.cfg.maml.inner_lr)
+        ctx.clone = maml.clone()
+
     def _hook_on_batch_forward(self, ctx):
-        # sub task
         batch = ctx.data_batch.to(ctx.device)
         if self.cfg.model.task.endswith('Regression'):
```
Comment (yxdyc) [inline federatedscope/gfl/trainer/mamltrainer.py:14]: This task-dependent condition statement is useful and we may add it into the GeneralTorchTrainer later

**Review by yxdyc — COMMENTED** [review]
LGTM, plz see the inline comments

File: federatedscope/gfl/trainer/mamltrainer.py (hunk 2 above)
Comment (DavdGao) [reply federatedscope/gfl/trainer/mamltrainer.py:4]: Currently the maml-based meta solution is only implemented in the contest branch. We'll discuss if it should be added into the master branch.

File: federatedscope/gfl/trainer/mamltrainer.py (hunk 1 above)
Comment (DavdGao) [reply federatedscope/gfl/trainer/mamltrainer.py:10]: Thanks, it is modified accordingly.

**Review by yxdyc — APPROVED** [review]

**Pull request merged**